pandas/core/groupby/generic.py

```diff
@@ -622,33 +625,27 @@ def value_counts(
         ids, val = ids[mask], val[mask]
 
         if bins is None:
-            lab, lev = algorithms.factorize(val, sort=True)
+            lab, lev = cast(
+                Tuple[np.ndarray, Index], algorithms.factorize(val, sort=True)
+            )
             llab = lambda lab, inc: lab[inc]
         else:
 
-            # lab is a Categorical with categories an IntervalIndex
-            lab = cut(Series(val), bins, include_lowest=True)
-            # error: "ndarray" has no attribute "cat"
-            lev = lab.cat.categories  # type: ignore[attr-defined]
-            # error: No overload variant of "take" of "_ArrayOrScalarCommon" matches
-            # argument types "Any", "bool", "Union[Any, float]"
-            lab = lev.take(  # type: ignore[call-overload]
-                # error: "ndarray" has no attribute "cat"
-                lab.cat.codes,  # type: ignore[attr-defined]
+            # labels is a Series with categories an IntervalIndex
+            labels: Series = cut(Series(val), bins, include_lowest=True)
```

Comment:
I'm not a fan of variable annotations where they should not be necessary. When an expression resolves to `Any` as is the case here since the return of `cut` is untyped, the variable annotation behaves just like a cast and allows us to lie to mypy (if we wanted too or unintentially). How difficult is it to add the type annotations to `pandas.core.reshape.tile.cut`?

Reply:
I will have a look on it on Tuesday and let you know what I find

Reply:
@simonjayhawkins ok, so it seems that `pandas.core.reshape.tile.cut` could work but would require additional overloads. The return value has a lot of dependencies on cut's parameters. Should I try to fix it in a separate PR given one of your previous comments?
> I have been trying to avoid adding overloads till the test framework is in place

I would also like to try to get this PR to the merge'able state, should I simply leave this ignore here for the time being (and potentially fix it in the PR adding the overloads)?

Reply:
> I would also like to try to get this PR to the merge'able state, should I simply leave this ignore here for the time being (and potentially fix it in the PR adding the overloads)?

sgtm.

because there is ongoing work to merge the MS Stubs into the pandas repository #45253 and add a test framework for the public api #45594, we need to be careful not to clash with that effort for now.

unfortunately #45594 which is IMO the most important aspect of the effort to merge the MS stubs has gone stale.

@FactorizeD I do appreciate the work done in this PR, but some of these ignores are better fixed after the MS Stubs are merged

The alternative maybe to open a precursor to this to bring in the types from the MS stubs for pandas.cut. I've not been following the progress of that and what the process is. @Dr-Irv?

Reply:
> The alternative maybe to open a precursor to this to bring in the types from the MS stubs for pandas.cut. I've not been following the progress of that and what the process is. @Dr-Irv?

The MS stubs were initially created by Microsoft by auto-generating them from pandas source.  At some point, the stub for `pd.cut()` had no type annotations and was in the repo, but has since disappeared.  Now, there is no stub at all, so that's something that will have to be added at some point.

Reply:
Ok, in this case I will leave it for now